### DISCUSSION TITLE
Problem retrieving the recipient endpoint

### DISCUSSION THREAD

**Discussion by jonathangraci** [OxalisCommunity/Oxalis-AS4#243] [General]
Hello everyone, I need some help,
when I try to send a document with Oxalis I get the following error:

    - Unable to fetch 'http://test-infra.peppol.at**http**/iso6523-actorid-upis... - No transmission is provided.

It's as if "**http**" is always inserted after the reference endpoint.

Can you help me understand where the problem is and how to solve it?

Thank you,
Jonathan.

**Comment by dladlk** [comment]
First guess - old Oxalis version, which does not support the change in SML NAPTR: https://www.oxalis.network/technical-information
Upgrade or switch to busdox

**Comment by jonathangraci** [comment]
Hi Dmytro,
we currently use version 4.0.1. In this version we are unable to use the "lookup.locator.class" property.

Is there a possibility to fix the problem in this version?

Thank you,
Jonathan.

**Comment by dladlk** [comment]
You can backport the fix (quite simple) to your version of peppol-lookup.jar of BdxlLocator class.

Not sure which version of peppol-lookup is in 4.0.1, but its source looks like (for 1.0.0 version):

https://github.com/OxalisCommunity/vefa-peppol/blob/87d5b79158b1555a858b3b52c43527dcdc2e876c/peppol-lookup/src/main/java/no/difi/vefa/peppol/lookup/locator/BdxlLocator.java#L127-L142

```
        // Simple stupid
        if ("^.*$".equals(regexp[1]))
            return regexp[2];
```

You should change it according to SML migration to be similar to https://github.com/OxalisCommunity/vefa-peppol/blob/5511ab68738bfbce2f78b1300fb4b54474584871/peppol-lookup/src/main/java/network/oxalis/vefa/peppol/lookup/locator/BdxlLocator.java#L184-L199

```
        // Simple stupid
        if (".*".equals(regexp[1]))
            return regexp[2];
```

because now NAPTR records return just .* instead of ^.*$ like before 15.09.2021...

Be aware, that last couple of years peppol-lookup changed package from no.difi.vefa to network.oxalis.vefa, so just modify this single line.

**Comment by aaron-kumar** [comment]
> Hi Dmytro, we currently use version 4.0.1. In this version we are unable to use the "lookup.locator.class" property. Is there a possibility to fix the problem in this version? Thank you, Jonathan.

@jonathangraci Sorry we can't provide support for any oxalis version prior than 6.x.x. Simple reason for this is that Oxalis version prior to 6.x.x are Non-compliant as per OpenPeppol specifications. 

Closing issue and changing it to discussion
